src/components/NewCommentForm/NewCommentForm.jsx

```diff
@@ -1,39 +1,109 @@
-import React from 'react';
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { addNewComment, getPostComments } from '../../api/api';
 import './NewCommentForm.scss';
 
-export const NewCommentForm = () => (
-  <form className="NewCommentForm">
-    <div className="form-field">
-      <input
-        type="text"
-        name="name"
-        placeholder="Your name"
-        className="NewCommentForm__input"
-      />
-    </div>
-
-    <div className="form-field">
-      <input
-        type="text"
-        name="email"
-        placeholder="Your email"
-        className="NewCommentForm__input"
-      />
-    </div>
-
-    <div className="form-field">
-      <textarea
-        name="body"
-        placeholder="Type comment here"
-        className="NewCommentForm__input"
-      />
-    </div>
-
-    <button
-      type="submit"
-      className="NewCommentForm__submit-button button"
+export const NewCommentForm = ({ selectedPostId, setPostComments }) => {
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [body, setBody] = useState('');
+  const [errors, setErrors] = useState({
+    name: false,
+    email: false,
+    body: false,
+  });
+
+  const validateCommentInputs = (field, input) => {
+    if (!input) {
+      setErrors({ [field]: true });
+
+      return false;
+    }
+
+    setErrors({ [field]: false });
+
+    return true;
+  };
+
+  const setDefaultValues = () => {
+    setName('');
+    setEmail('');
+    setBody('');
+  };
+
+  const handleSubmit = async(e) => {
+    e.preventDefault();
+
+    if (!validateCommentInputs('name', name)
+    || !validateCommentInputs('email', email)
+    || !validateCommentInputs('body', body)) {
+      return;
+    }
+
+    const newComment = {
+      name,
+      email,
+      body,
+      postId: selectedPostId,
+      createdAt: new Date(),
+    };
+
+    await addNewComment(newComment);
+    getPostComments(selectedPostId).then(setPostComments);
+    setDefaultValues();
+  };
+
+  return (
+    <form
+      className="NewCommentForm"
+      onSubmit={e => handleSubmit(e)}
     >
-      Add a comment
-    </button>
-  </form>
-);
+      <div className="form-field">
+        <input
+          type="text"
+          name="name"
+          placeholder="Your name"
+          value={name}
+          className="NewCommentForm__input"
+          onChange={e => setName(e.target.value)}
+        />
+      </div>
+      {errors.name && <h3>Please, enter your name</h3>}
+
+      <div className="form-field">
+        <input
+          type="text"
+          name="email"
+          placeholder="Your email"
+          value={email}
+          className="NewCommentForm__input"
+          onChange={e => setEmail(e.target.value)}
```

Comment:
```suggestion
          onChange={setEmail}
```
А target.value доставай внутри функции